### ISSUE TITLE
account:balance incorrect

### ISSUE THREAD

**Issue by terrancewong** [iron-fish/ironfish#1293]
### What happened?

on machine **A**, with certain account **X** as default. miner connected to this machine. 
```
$ ironfish status 
Version              0.1.31 @ src
Node                 STARTED
Memory               Heap: 56.51 MiB / 84.36 MiB (67.0%), RSS: 1.21 GiB (5.2%), Free: 20.15 GiB (14.2%)
P2P Network          CONNECTED - In: 126.36 KB/s, Out: 41.33 KB/s, peers 48
Mining               STARTED - 2 miners, 5 mined
Mem Pool             3 tx
Syncer               IDLE - avg time to add block 105.09 ms
Blockchain           SYNCED @ HEAD 00000000000fe1e46a1d949a9efb2c56393c66110872db21909f32f6a50d17fe (10013)
Telemetry            STOPPED
Workers              STARTED - 0 -> 1 / 5 - -0.13 jobs Δ, 35.47 jobs/s

$ ironfish accounts:balance
Account - <X> (<hex>)

The balance is: $IRON 92.69999949 ($ORE 9,269,999,949)
Amount available to spend: $IRON....
```

On machine **B**, also with certain account **X** as default, another account **Y** was imported. **X** transferred some $IRON to **Y**, so total amount should be less than 100.00000xxx

status said
```
Version              0.1.31 @ src
Node                 STARTED
Memory               Heap: 58.19 MiB / 78.86 MiB (73.8%), RSS: 582.06 MiB (1.8%), Free: 11.66 GiB (62.1%)
P2P Network          CONNECTED - In: 2.11 KB/s, Out: 4.59 KB/s, peers 50    
Mining               STARTED - 0 miners, 0 mined
Mem Pool             9 tx  
Syncer               IDLE - avg time to add block 38.04 ms                                                
Blockchain           SYNCED @ HEAD 00000000001c33b2fa9c21688d8b66fec6e4b2aa6fea69657c6e35b8aeff74a6 (10015)   
Telemetry            STOPPED
Workers              STARTED - 0 -> 0 / 5 - 0 jobs Δ, 11.11 jobs/s  
```

```
$ ironfish accounts:balance --verbose --verbose
Connecting to 0.0.0.0:9023
Account - <X> (<hex>)

The balance is: $IRON 185.89999903 ($ORE 18,589,999,903)
Amount available to spend: $IRON ...
```

accounts:balance shows balance larger than real value.

### Version

v0.1.31

### Debug output

```shell
Database in use, skipping output that requires database.
Stop the node and run the debug command again to show full output.

Iron Fish version       0.1.31 @ src
Iron Fish library       0.0.9 @ src
Operating system        Linux x64
CPU model(s)            Intel(R) Xeon(R) CPU E3-1245 v6 @ 3.70GHz
CPU threads             8
RAM total               30.78 GiB
Node version            v16.13.1
ironfish in PATH        false
Telemetry enabled       false
```
```


### Relevant log output

```shell
NA
```

### Graffiti

bh9sqoCqIR3rDFstCCCw1g

**Comment by terrancewong** [comment]
Further info:
Now machine **A** also shows incorrect balance

status says 6 blocks mined
```
Version              0.1.31 @ src
Node                 STARTED
Memory               Heap: 53.03 MiB / 85.36 MiB (62.1%), RSS: 1.25 GiB (5.3%), Free: 20.14 GiB (14.2%)
P2P Network          CONNECTED - In: 2 B/s, Out: 4.98 KB/s, peers 48        
Mining               STARTED - 2 miners, 6 mined
Mem Pool             6 tx 
Syncer               IDLE - avg time to add block 123.67 ms
Blockchain           SYNCED @ HEAD 000000000002b7d0b3805bf545c6cf820cf05e3a053785faa1daa8087ae2125a (10029)
Telemetry            STOPPED
Workers              STARTED - 0 -> 0 / 5 - 0 jobs Δ, 0 jobs/s     
```

`accounts:balance` shows much more, (some $IRON already transferred out) 
```
ironfish accounts:balance
Account - <X> (<hex checksum>)

The balance is: $IRON 145.99999889 ($ORE 14,599,999,889)
Amount available to spend: $IRON 145.99999889 ($ORE 14,599,999,889)
```

a `rescan` does not help
```
$ ironfish accounts:rescan 
Rescanning Transactions... Scanning Complete
$ ironfish accounts:balance
Account - <X> (<hex>)

The balance is: $IRON 145.99999889 ($ORE 14,599,999,889)
Amount available to spend: $IRON 145.99999889 ($ORE 14,599,999,889)
```

**Comment by lwisne** [comment]
Did this ever resolve? Can look into it further if not.

**Comment by hughy** [comment]
The Wallet 2.0 refactor released in [v0.1.48](https://github.com/iron-fish/ironfish/releases/tag/v0.1.48) included significant changes to the way that account balances are stored and calculated.

Please open a new issue if you run into this again!
